lms/static/scripts/frontend_apps/components/dashboard/test/AssignmentActivity-test.js

```diff
@@ -171,6 +189,114 @@ describe('AssignmentActivity', () => {
     });
   });
 
+  context('when filters are set', () => {
+    function setCurrentURL(url) {
+      history.replaceState(null, '', url);
+    }
+
+    beforeEach(() => {
+      setCurrentURL('?');
+    });
+
+    it('initializes expected filters', () => {
+      setCurrentURL('?student_id=1&student_id=2');
+
+      const wrapper = createComponent();
+      const filters = wrapper.find('DashboardActivityFilters');
+
+      // Active course and assignment are set from the route
+      assert.deepEqual(
+        filters.prop('courses').activeItem,
+        activeAssignment.course,
+      );
+      assert.deepEqual(
+        filters.prop('assignments').activeItem,
+        activeAssignment,
+      );
+      // Students are set from the query
+      assert.deepEqual(filters.prop('students').selectedIds, ['1', '2']);
+
+      // Selected filters are propagated when loading assignment metrics
+      assert.calledWith(fakeUseAPIFetch.lastCall, sinon.match.string, {
+        h_userid: ['1', '2'],
+        assignment_id: '123',
+        public_id: undefined,
```

Comment:
We really should replace `public_id` with `org_id` or similar. Otherwise it is unclear what kind of entity it refers to.

Reply:
Let me track that https://github.com/hypothesis/lms/issues/6541